src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -294,6 +296,20 @@ private FolioExecutionContext prepareContextForTenant(String tenantId, FolioExec
     return new DefaultFolioExecutionContext(folioModuleMetadata, context.getOkapiHeaders());
   }
 
+  public void deleteShadowUsers() {
+    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    List<UserTenantEntity> userTenantEntities = userTenantRepository.getByUserIdAndIsPrimaryFalse().orElse(null);
+    if (Objects.nonNull(userTenantEntities)) {
+      Map<String, List<UserTenantEntity>> listMap = userTenantEntities.stream().collect(Collectors.groupingBy(userTenantEntity -> userTenantEntity.getTenant().getId()));
+      listMap.forEach((key, value) -> {
```

Comment:
please add logs of what user ids from what which tenants we are deleting

Reply:
fixed